tools/tpm2_readpublic.c

```diff
@@ -92,8 +95,16 @@ static int read_public_and_save(TSS2_SYS_CONTEXT *sapi_context) {
     }
     tpm2_tool_output("\n");
 
-    return files_save_bytes_to_file(ctx.outFilePath, (UINT8 *) &public,
-            sizeof(public));
+    pubkey_format format = pubkey_format_tss;
```

Comment:
set this in the static global, then you can just update it in the option handler, no need for checking if the f flag is set.